PayPal/src/main/java/com/braintreepayments/api/PayPalLauncher.java

```diff
@@ -1,93 +1,95 @@
+        } else if (request instanceof BrowserSwitchPendingRequest.Started) {
+            return new PayPalPendingRequest.Started(new PayPalBrowserSwitchRequest(
+                    (BrowserSwitchPendingRequest.Started) request));
         }
+        return new PayPalPendingRequest.Failure(new BraintreeException("An unexpected error occurred"));
     }
 
     /**
      * Captures and delivers the result of a the browser-based PayPal authentication flow.
      * <p>
      * For most integrations, this method should be invoked in the onResume method of the Activity
-     * used to invoke {@link PayPalLauncher#launch(FragmentActivity, PayPalPaymentAuthRequestParams)}.
+     * used to invoke
+     * {@link PayPalLauncher#launch(ComponentActivity, PayPalPaymentAuthRequest.ReadyToLaunch)}.
      * <p>
      * If the Activity used to launch the PayPal flow has is configured with
      * android:launchMode="singleTop", this method should be invoked in the onNewIntent method of
-     * the Activity, after invoking setIntent(intent).
-     * <p>
-     * This method will deliver a {@link PayPalPaymentAuthResult} to the
-     * {@link PayPalLauncherCallback} used to instantiate this class. The
-     * {@link PayPalPaymentAuthResult} should be passed to
-     * {@link PayPalClient#tokenize(PayPalPaymentAuthResult, PayPalTokenizeCallback)}
+     * the Activity.
      *
-     * @param context the context used to check for pending results
-     * @param intent  the intent to return to your application containing a deep link result from
-     *                the PayPal browser flow
+     * @param pendingRequest the {@link PayPalPendingRequest.Started} stored after successfully
+     *                       invoking {@link PayPalLauncher#launch(ComponentActivity, PayPalPaymentAuthRequest.ReadyToLaunch)}
+     * @param intent         the intent to return to your application containing a deep link result
+     *                       from the PayPal browser flow
+     * @return a {@link PayPalPaymentAuthResult} that should be passed to
+     * {@link PayPalClient#tokenize(PayPalPaymentAuthResult, PayPalTokenizeCallback)} to complete
+     * the PayPal payment flow. Returns null if the user closed the browser to cancel the payment
+     * flow, or returned to the app without completing the PayPal authentication flow.
      */
-    public void handleReturnToAppFromBrowser(@NonNull Context context, @NonNull Intent intent) {
-        BrowserSwitchResult result = browserSwitchClient.parseResult(context, PAYPAL, intent);
+    @Nullable
+    public PayPalPaymentAuthResult handleReturnToAppFromBrowser(
+            @NonNull PayPalPendingRequest.Started pendingRequest, @NonNull Intent intent) {
+        BrowserSwitchResult result = browserSwitchClient.parseResult(pendingRequest.getRequest().getBrowserSwitchPendingRequest(), intent);
+        PayPalPaymentAuthResult paymentAuthResult = null;
```

Comment:
Take it or leave it comment - curious here why return `null` versus throw an error for the user-did-not-complete-flow case?

Reply:
This is to avoid being overly opinionated in the SDK - the browser switch method under the hood essentially just checks to see if there is a matching result for the intent and pending request passed in, and if there is, returns a success. If there isn't, the merchant can infer that the user didn't complete the flow based on their own setup and the request they passed in, but the SDK can't be certain that it means the user didn't complete the flow, because the merchant could have passed any request in at any point

Reply:
Got it, thank you!